app/Repositories/IndependentActivity/IndependentActivityRepository.php

```diff
@@ -831,4 +833,53 @@ public function updateIndex($independentActivity, $index): string
         $independentActivity->update(['indexing' => $index]);
         return 'Library status changed successfully!';
     }
+
+    /**
+     * Update Indexes for independent activities and related models
+     * @param $independentActivity
+     * @param $playlist
+     * @param $token
+     * @return string
+     * 
+     */
+    public function copyToPlaylist( $independentActivity, $playlist, $token)
+    {
+        $h5p_content = $independentActivity->h5p_content;
+        if ($h5p_content) {
+            $h5p_content = $h5p_content->replicate(); // replicate the all data of original activity h5pContent relation
+            $h5p_content->user_id = get_user_id_by_token($token); // just update the user id which is performing the cloning
+            $h5p_content->save(); // this will return true, then we can get id of h5pContent
+        }
+        $newH5pContent = $h5p_content->id ?? null;
+
+        // copy the content data if exist
+        $this->copy_content_data($independentActivity->h5p_content_id, $newH5pContent);
+
+        $new_thumb_url = clone_thumbnail($independentActivity->thumb_url, "activities");
+        $activity_data = [
+            'title' => $independentActivity->title,
+            'type' => $independentActivity->type,
+            'content' => $independentActivity->content,
+            'playlist_id' => $playlist->id,
+            'order' => $this->getOrder($playlist->id) + 1,
+            'h5p_content_id' => $newH5pContent, // set if new h5pContent created
+            'thumb_url' => $new_thumb_url,
+            'subject_id' => $independentActivity->subject_id,
```

Comment:
There are no longer any "subject_id" and "education_level_id" fields for independentActivity.
Instead independentActivity has many to many relation with subjects and education levels.

Also even the existing activity no longer use "subject_id" and "education_level_id" fields.
It now also has many to many relation with subjects and education levels.

Please update the code accordingly.